mesonbuild/modules/python.py

```diff
@@ -727,10 +727,27 @@ def find_installation(self, state: 'ModuleState', args: T.Tuple[T.Optional[str]]
         found_modules: T.List[str] = []
         missing_modules: T.List[str] = []
         if python.found() and want_modules:
+            # Python 3.8.x or later require add_dll_directory() to be called on Windows if
+            # the needed modules require external DLLs that are not bundled with the modules.
+            # Simplify things by calling add_dll_directory() on the paths in %PATH%
+            add_paths_cmd = None
+            if hasattr(os, 'add_dll_directory'):
+                paths = os.environ['PATH'].split(os.pathsep)
+                add_paths_cmds = []
+                for path in paths:
+                    if path != '' and os.path.isdir(path):
+                        add_paths_cmds.append("os.add_dll_directory(r'%s')" % path.strip(os.sep))
```

Comment:
Why does this need to have the path stripped of leading `\\` ?

Also:
```suggestion
                        add_paths_cmds.append('os.add_dll_directory({!r}).format(path.strip(os.sep)))
```

Reply:
Oh, yeah, UNC paths...

Thanks for the notes.

Reply:
Okay. Why does it need to be stripped of trailing \\ either?

Reply:
Hi @eli-schwartz,

Sometimes people may have a trailing `\` in one or more of the directories that is included in `%PATH%` (you never know how people setup their `%PATH%` envvar), and that will cause trouble, one will get a Python error for an unescaped `\` when calling `os.add_dll_directory()`, so that `.rstrip(os.sep)` became needed.

Reply:
This doesn't seem correct to me at all.

```python
>>> path = 'C:\\foo\\bar\\'
>>> print(path)
C:\foo\bar\
>>> func = 'print({!r})'.format(path)
>>> func
"print('C:\\\\foo\\\\bar\\\\')"
>>> subprocess.run[sys.executable, '-c', func])
C:\foo\bar\
CompletedProcess(args=['/usr/bin/python', '-c', "print('C:\\\\foo\\\\bar\\\\')"], returncode=0)
```

Reply:
Although that is probably just a bug which my review suggestion to use `{!r}` has solved, since the repr of a string is guaranteed to be formatted correctly for reuse, very unlike the str() surrounded by quotes.

Reply:
OK, I will give it a shot.  Maybe 3-5 minutes?